src/PointwiseFunctions/ScalarGaussBonnet/ScalarMomentum.hpp

```diff
@@ -0,0 +1,77 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "Elliptic/Systems/ScalarGaussBonnet/Tags.hpp"
+#include "Elliptic/Systems/Xcts/Tags.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+
+namespace sgb {
+/// @{
+/*!
+ * \brief Compute the momentum$\Pi$ assuming quasi-stationarity in sGB gravity.
+ *
+ * This expression can be obtained by simply demanding $\partial_t \Psi = 0$,
+ * yielding
+ *
+ * \begin{equation}
+ * \Pi \equiv -n^{a} \partial_a = \alpha^{-1} \beta^{i} \partial_i \Psi
+ * \end{equation}
+ */
+
+void scalar_momentum(gsl::not_null<Scalar<DataVector>*> result,
+                     const tnsr::i<DataVector, 3, Frame::Inertial>& deriv,
+                     const tnsr::I<DataVector, 3>& shift,
+                     const Scalar<DataVector>& lapse);
+
+Scalar<DataVector> scalar_momentum(
+    const tnsr::i<DataVector, 3, Frame::Inertial>& deriv,
+    const tnsr::I<DataVector, 3>& shift, const Scalar<DataVector>& lapse);
+/// @{
+
+namespace Tags {
+
+/*!
+ * \brief Re-compute the momentum Pi, assuming quasi-stationarity.
+ */
+struct PiCompute : ::CurvedScalarWave::Tags::Pi, db::ComputeTag {
```

Comment:
(optional) If you want you can avoid the two nearly identical structs by templating on `ShiftTag` and `OutputTag`:

```cpp
template <typename ShiftTag, typename OutputTag>
struct PiCompute : OutputTag, db::ComputeTag {
  // ...
  using argument_tags = tmpl::list<ShiftTag, ...>;
```